entanglement_forging/core/wrappers/entanglement_forged_vqe.py

```diff
@@ -75,15 +107,46 @@ def __init__(self, ansatz, bitstrings, config, forged_operator, classical_energi
         self._paramsets_each_iteration = []
         self._schmidt_coeffs_each_iteration_each_paramset = []
         self._zero_noise_extrap = config.zero_noise_extrap
-        self.bitstrings = bitstrings
-        self._bitstrings_s = np.asarray(bitstrings)
+
+        self.bitstrings_u = bitstrings_u
+        self._bitstrings_s_u = np.asarray(bitstrings_u)
+
+        # Prevent unnecessary duplication of ansatz calculation if U=V
+        if bitstrings_v == bitstrings_u:
+            self.bitstrings_v = []
+        else:
+            self.bitstrings_v = bitstrings_v
+
+        # Make circuits which prepare states U|b_n_u> and U|phi^p_nm_u>
         (
-            self._tensor_prep_circuits,
-            self._superpos_prep_circuits,
-        ) = make_stateprep_circuits(bitstrings, config.fix_first_bitstring)
+            self._tensor_prep_circuits_u,
+            self._superpos_prep_circuits_u,
+            superpos_coeffs_u,
```

Comment:
More descriptive variable name. Something like "hybrid_superpos_coeffs" and include a comment.